### PR TITLE
Added documentation and removed unecessary files

### PR DESCRIPTION
**Pull request by gneatgeek**
### Summary
- Added documentation for the mixins
- Removed unnecessary files
- Removed unnecessary dependencies

### CODE REVIEW

**Comment by gneatgeek** [comment]
This works towards completed #2

**Comment by Matt-Butler** [comment]
Can you please update `docs/README.md`?

**Comment by Matt-Butler** [comment]
+1 after addressing comments

File: package.json [hunk 1]
```diff
@@ -40,9 +39,7 @@
     "stylelint": "^7.1.0",
     "stylelint-config-sass-guidelines": "^1.1.0",
     "terra-application": "^0.2.0",
```
Comment (Matt-Butler) [inline package.json]: delete this too please. 

Avoid circular dependency

File: docs/README.md [hunk 2]
```diff
@@ -1,3 +1,11 @@
 # Terra Mixins Documentation
 
 - [Terra Mixins](terra-mixins.md)
+
+Themeable variables:
+
+- [Themeable Variables](themeable-variables.md)
+
+Additional modifiers:
```
Comment (bjankord) [inline docs/README.md]: I'm not sure additional modifiers makes sense as a heading here. Let's discuss this some more today.

**Pull request merged**